File: 111-minimum-depth-of-binary-tree/111-minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if (root == nullptr) {
            return 0;
        }
        if (root->left == nullptr && root->right == nullptr) {
            return 1;
        }
        int lDepth = minDepth(root->left);
        int rDepth = minDepth(root->right);
        if (lDepth && rDepth) {
            return 1 + min(lDepth, rDepth);
        } else if (lDepth) {
            return 1 + lDepth;
        } else {
            return 1 + rDepth;
        }
        //return 1 + min(minDepth(root->left), minDepth(root->right));
    }
};

File: 111-minimum-depth-of-binary-tree/111-minimum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {    
    if (root == null) {
        return 0;
    }
    root.level = 1;
    let queue = [root];
    while (queue.length > 0) {
        let currNode = queue.shift();
        
        if (currNode.left != null) {
            currNode.left.level = currNode.level + 1;
            queue.push(currNode.left);
        }
        if (currNode.right != null) {
            currNode.right.level = currNode.level + 1;
            queue.push(currNode.right);
        }
        if (currNode.left == null && currNode.right == null) {
            return currNode.level;
        } 
        
    }
    throw ("Something went wrong");
};